katdal/auth.py

```diff
@@ -0,0 +1,169 @@
+"""Authentication with the MeerKAT archive.
+
+This provides integration with OpenID Connect authentication to authenticate to
+a proxy in front of an S3 service, rather than using S3 secret key
+authentication. It does not hardcode any MeerKAT-specific addresses, so could
+be used elsewhere. The only requirement is that the proxy provides a
+X-Katdal-Authentication-Provider header with the URL of the OpenID provider, and
+that the provider supports out-of-band transfer of the code.
+
+When authentication is required, it gives an URL to the user. The user opens
+the link in a browser (possibly on a separate machine) and obtains an
+authorization code, which they paste back into the terminal running katdal.
+"""
+
+from __future__ import print_function
+import json
+import requests
+import threading
+import urlparse
+import logging
+
+import oic
```

Comment:
Out of curiosity, why `oic` and not `pyoidc`?

Reply:
For some reason the package is pyoidc but the module is oic. One of those annoying ones.